### PR TITLE
Update tests to support v0.10.0 pathfinder node

### LINKED ISSUE #36: Implement starknet_getNonce
starknet_getNonce API doc [here](https://github.com/starkware-libs/starknet-specs/blob/master/api/starknet_api_openrpc.json#L628).

The boiler plate code has been implemented as part of #17. However, the method was unimplemented at the time of development.

### LINKED ISSUE #57: Update the tests to target v0.10.0 pathfinder node and fix the tests
- [x] Modify [here](https://github.com/gabsn/starknet.dart/blob/f7156603f80625d9967b84ed413f27909de99c41/test/utils.dart#L5) to target the pathfinder v0.10.0 node instead
- [ ] Modify [here](https://github.com/gabsn/starknet.dart/blob/f7156603f80625d9967b84ed413f27909de99c41/test/utils.dart#L19) to target the pathfinder v0.10.0 node instead
- [x] Fix the tests to make sure the SDK actually works with pathfinder v0.10.0 specs

### PR DESCRIPTION
**Pull request by prajwal27**
Mainly Closes #57 
Also closes below ones as these are part of the above main issue.
#36 
#37 
#38 
#40 
#42

### CODE REVIEW

File: test/provider/read_provider_test.dart [hunk 1]
```diff
@@ -19,70 +19,180 @@ void main() {
...
-                      'method \'starknet_getBlockWithTxHashes\' not found'));
+              expect(error.code, -32602);
+              expect(error.message, contains("invalid value: integer `-1`"));
             });
       });
     });
 
     group('call', () {
-      test('calls a read-only method with empty calldata', () async {
+      // TODO: needs to be updated for v0.10.0 specs url
```
Comment (gabsn) [inline test/provider/read_provider_test.dart:51]: I think we should fix the `call` tests before merging

File: test/provider/read_provider_test.dart [hunk 2]
```diff
@@ -19,70 +19,180 @@ void main() {
...
+              expect(error.code, 21);
+              expect(error.message, contains("Invalid message selector"));
+            },
+            result: (result) => fail("Should fail"));
       });
     });
 
     group('getStorageAt', () {
-      test('returns the ERC20_symbol value for a ERC20 contract', () async {
+      // TODO: needs to be updated for v0.10.0 specs url
```
Comment (gabsn) [inline test/provider/read_provider_test.dart:165]: Same, let's fix the tests for `getStorageAt`

**Review by gabsn — CHANGES_REQUESTED** [review]
Let's keep the tests for `call` and `getStorageAt`

File: test/provider/read_provider_test.dart (hunk 1 above)
Comment (prajwal27) [reply test/provider/read_provider_test.dart:51]: I have commented out the old tests and added new ones.

File: test/provider/read_provider_test.dart (hunk 2 above)
Comment (prajwal27) [reply test/provider/read_provider_test.dart:165]: Same, I have commented out the old tests and added new ones. It's mainly the old tests weren't working with `v010PathfinderGoerliTestnetUri`

**Pull request merged**